src/Sources/Condensation.jl

```diff
@@ -12,37 +12,23 @@ export get_cond_evap
 
 
 """
-    get_cond_evap(s::FT, par::NamedTuple)
+    get_cond_evap(pdists, s::FT, ξ::FT)
 
-    's' - supersaturation
-    `par` - ODE parameters, a NamedTuple containing a list of ParticleDistributions and the condensation coefficient ξ
+    'pdists` - local particle size distributions
+    's' - spatially-varying supersaturation
+    'ξ' - spatially-varying condensation coefficient (T, P dependent)
 Returns the rate of change of all prognostic moments due to condensation and evaporation (without ventilation effects)
 based on the equation dg(x) / dt = -3ξs d/dx(x^{1/3} g(x))
 """
-function get_cond_evap(s::FT, par::NamedTuple) where {FT <: Real}
-    ξ = par.ξ
-    n_dist = length(par.pdists)
-    NProgMoms = [nparams(dist) for dist in par.pdists]
-
-    # build diagnostic moments
-    mom = zeros(FT, sum(NProgMoms))
-    for i in 1:n_dist
-        for j in 2:NProgMoms[i]
-            ind = get_dist_moment_ind(NProgMoms, i, j)
-            mom[ind] = moment(par.pdists[i], FT(j - 1 - 2 / 3))
+function get_cond_evap(pdists::NTuple{N, PrimitiveParticleDistribution{FT}}, s::FT, ξ::FT) where {N, FT <: Real}
+    # build diagnostic moments & compute rate of change
+    cond_evap_int = map(pdists) do pdist
+        map(1:nparams(pdist)) do j
```

Comment:
```suggestion
        ntuple(nparams(pdist)) do j
```
This will ensure that the inner type is a tuple. I think the inner type may otherwise be a vector:
```julia
julia> map(1:3) do i
       i
       end
3-element Vector{Int64}:
 1
 2
 3
```